planner/core/exhaust_physical_plans.go

```diff
@@ -1704,7 +1704,10 @@ func (p *LogicalProjection) exhaustPhysicalPlans(prop *property.PhysicalProperty
 }
 
 func (lt *LogicalTopN) getPhysTopN(prop *property.PhysicalProperty) []PhysicalPlan {
-	allTaskTypes := prop.GetAllPossibleChildTaskTypes()
```

Comment:
Please add a method like `LogicalAggregation.canPushToCop()` to check if it is valid to push the `TopN` down.
For simplifying, we can push it only when its child is `DataSource`.
If we don't set the limitation, some errors may occur:
```
CREATE TABLE `t` (
  `a` bigint(20) unsigned DEFAULT NULL
);

mysql> explain select /*+ TOPN_TO_COP() */ a from t where mod(a, 2) order by a limit 1;
ERROR 1815 (HY000): Internal : Can't find a proper physical plan for this query
```

Reply:
PTAL